src/multikey.rs

```diff
@@ -173,29 +173,32 @@ impl Multisecret {
         let tail =
             skip_prefix(suffix, ED25519_SUFFIX).ok_or_else(|| DecodeLegacyError::UnknownSuffix)?;
 
-        if data.len() != SECRETKEYBYTES {
```

Comment:
I think this code was and is broken; the legacy encoding isn't expected to be SECRETKEYBYTES long. I'd say we replace the entire body of this function with a call to `Keypair::from_base64`

Reply:
Done